dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/OmegaRef/BaseClasses_INIT.mo

```diff
@@ -15,6 +15,116 @@ within Dynawo.Electrical.Machines.OmegaRef;
+    Types.PerUnit MdPPu "Direct axis mutual inductance in pu";
+    Types.PerUnit MqPPu "Quadrature axis mutual inductance in pu";
+    Types.PerUnit LdPPu "Direct axis stator leakage in pu";
+    Types.PerUnit LqPPu "Quadrature axis stator leakage in pu";
+    Types.PerUnit RaPPu "Armature resistance in pu";
+    Types.ComplexCurrentPu i0PuNom "Complex current at terminal in pu (base UNom, SNom)";
+
+    // Intermediate values calculated in the while loop to determine the rotor angle
+    Types.PerUnit XqPPu "Quadrature axis reactance in pu";
+    Types.PerUnit MsalPu "Constant difference between direct and quadrature axis saturated mutual inductances in pu";
+    Types.Angle Theta0 "Start value of rotor angle: angle between machine rotor frame and port phasor frame";
+    Types.VoltageModulePu Ud0PuNom "Start value of voltage of direct axis in pu";
+    Types.VoltageModulePu Uq0PuNom "Start value of voltage of quadrature axis in pu";
+    Types.CurrentModulePu Id0PuNom "Start value of current of direct axis in pu";
+    Types.CurrentModulePu Iq0PuNom "Start value of current of quadrature axis in pu";
+    Types.PerUnit LambdaAD0PuNom "Start value of common flux of direct axis in pu";
+    Types.PerUnit LambdaAQ0PuNom "Start value of common flux of quadrature axis in pu";
+    Types.PerUnit LambdaAirGap0PuNom "Start value of total air gap flux in pu";
+    Types.PerUnit Mds0PuNom "Start value of direct axis saturated mutual inductance in the case when the total air gap flux is aligned on the direct axis in pu";
+    Types.PerUnit Mqs0PuNom "Start value of quadrature axis saturated mutual inductance in the case when the total air gap flux is aligned on the quadrature axis in pu";
+    Types.PerUnit Cos2Eta0 "Start value of the common flux of direct axis contribution to the total air gap flux in pu";
+    Types.PerUnit Sin2Eta0 "Start value of the common flux of quadrature axis contribution to the total air gap flux in pu";
+    Types.PerUnit Mi0PuNom "Start value of intermediate axis saturated mutual inductance in pu";
+    Types.PerUnit MdSat0PPuNom "Start value of direct axis saturated mutual inductance in pu";
+    Types.PerUnit MqSat0PPuNom "Start value of quadrature axis saturated mutual inductance in pu";
+
+    // Intermediate values dealing with the algorithm loop condition
+    Types.PerUnit MdSat0PPuNomSave "Previous value for MdSat0PPuNom";
+    Types.PerUnit MqSat0PPuNomSave "Previous value for MqSat0PPuNom";
+    Types.PerUnit deltaMdSat0PPuNom "Variation of MdSat0PPuNom during the current iteration";
+    Types.PerUnit deltaMqSat0PPuNom "Variation of MqSat0PPuNom during the current iteration";
+    Integer nbIterations "Current number of iterations in the fix point algorithm";
+    Boolean iterate "Indicates if the while loop should continue";
+
+  algorithm
+    // Initial values calculations
+    i0PuNom := Complex(- PNomAlt, sqrt(SNom ^ 2 - PNomAlt ^ 2)) / SNom;
```

Comment:
What about convention here? 
I suppose that if we are in receptor convention, it should also be - sqrt().

Reply:
No because it's iPu and not sPu. As we have sPu = - Complex(PNom, QNom) = uPu * conj(iPu) and uPu = 1, it leads to iPu = conj(sPu) = Complex(PNom, -QNom).

Reply:
And we are in receptor convention